projects/dc2903a/src/examples/basic_example/basic_example.c

```diff
@@ -0,0 +1,144 @@
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "basic_example.h"
+#include "ltc2672.h"
+#include "common_data.h"
+#include "no_os_print_log.h"
+#include "no_os_delay.h"
+#include "no_os_util.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/***************************************************************************//**
+ * @brief basic example main execution.
+ *
+ * @return ret - Result of the example execution.
+*******************************************************************************/
+int basic_example_main()
+{
+	int i, ret, num_currents, toggle_times = 4;
+	bool toggle_flag = false;
+	float currents_to_set[] = {.21, .41, 1.21, 1.41}; // current to set in mA
+	float currents_to_toggle[] = {0.4, 0.1};
+	struct ltc2672_dev *ltc2672_desc;
+
+	num_currents = NO_OS_ARRAY_SIZE(currents_to_set);
+
+	/* Initialize device */
+	ret = ltc2672_init(&ltc2672_desc, &ltc2672_ip);
+	if (ret)
+		return ret;
+
+	/* Continuously configure the part and measure pins with multimeter for verification */
+	while (1) {
+		ret = ltc2672_set_span_all_channels(ltc2672_desc,
+						    LTC2672_50VREF); // max current of 3.125mA
+		if (ret)
+			return ret;
+
+		/* Print Span set to all channels */
+		pr_info("All DAC channels have span config of 3.125mA\n");
+
+		for (i = 0; i < num_currents; i++) {
+			ret = ltc2672_set_current_all_channels(ltc2672_desc, currents_to_set[i]);
+			if (ret)
+				return ret;
```

Comment:
same comment about memory leak as above, return only after ltc2672_remove is called